pkg/monitor/cluster/nodeconditions.go

```diff
@@ -23,27 +27,35 @@ func (mon *Monitor) emitNodeConditions(ctx context.Context) error {
 		return err
 	}
 
+	spotInstances := mon.getSpotInstances(ctx)
+
 	mon.emitGauge("node.count", int64(len(ns.Items)), nil)
 
 	for _, n := range ns.Items {
+		nodeConditions := make(map[string]string)
+		nodeConditions["nodeName"] = n.Name
+
 		for _, c := range n.Status.Conditions {
 			if c.Status == nodeConditionsExpected[c.Type] {
 				continue
 			}
 
-			mon.emitGauge("node.conditions", 1, map[string]string{
-				"nodeName": n.Name,
-				"status":   string(c.Status),
-				"type":     string(c.Type),
-			})
+			_, isSpotInstance := spotInstances[n.Name]
+			nodeConditions["spotInstance"] = strconv.FormatBool(isSpotInstance)
+
+			nodeConditions["status"] = string(c.Status)
+			nodeConditions["type"] = string(c.Type)
+
+			mon.emitGauge("node.conditions", 1, nodeConditions)
```

Comment:
IMO, this looks a bit more readable than having `nodeConditions` map all around the place. Or do I miss a reason for constructing `nodeConditions` this way?

```suggestion
			mon.emitGauge("node.conditions", 1, map[string]string{
				"nodeName":     n.Name,
				"status":       string(c.Status),
				"type":         string(c.Type),
				"spotInstance": isSpotInstance,
			})
```

Reply:
Also why do we need `strconv.FormatBool`? This will mean that bools will be serialised as strings in metrics.

Reply:
I can remove the map, it was from some previous refactoring. we convert it to a string to match the maps definition of [string]string